cross_list_courses/static/cross_list_courses/js/controllers/ListController.js

```diff
@@ -110,14 +121,87 @@
                 }).finally(function cleanupAfterDelete() {
                     // always reload, in case the failure was a 404
                     $scope.dtInstance.reloadData();
-                    $scope.operationInProgress = null;
                 });
 
             return promise;
         };
+        $scope.search = function() {
+            $scope.clearMessages();
+            $scope.setOperationInProgress('search');
+            // Call within timeout to prevent
+            // https://docs.angularjs.org/error/$rootScope/inprog?p0=$apply
+            $timeout(function () {
+                $scope.dtInstance.reloadData();
+            }, 0);
+        };
+        $scope.setOperationInProgress = function(newSetting) {
+            // disable datatable and show progress bar if newSetting is truthy
+            $timeout(function() {
+                // notify UI to start/stop showing in-progress messaging
+                $scope.operationInProgress = newSetting;
+                // enable/disable interactive data table elements
+                $scope.setDataTableInteraction(!newSetting);
```

Comment:
Right -- it treats any string (the name of the operation) as a boolean for the purposes of determining whether or not to allow interaction with the datatable. That is, if it is a non-zero-length string, Javascript considers it truthy. Perhaps it would be best to explicitly cast it to a boolean using `!!newSetting` first? Or just some clarification in an inline comment?